spikeinterface/toolkit/preprocessing/resample.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import scipy.signal
+
+from .basepreprocessor import (
+    BasePreprocessor,
+    BasePreprocessorSegment,
+)
+from .filter import fix_dtype
+
+try:
+    from scipy import special, signal
+
+    HAVE_RR = True
+except ImportError:
+    HAVE_RR = False
+
+
+class ResampleRecording(BasePreprocessor):
+    """
+    Generic resample class based the previous SpikeToolkits API.
+
+    If the resampling rate is multiple of the sampling rate, the faster
+    scipy decimate function is used.
+
+    Parameters
+    ----------
+    recording: Recording
+        The recording extractor to be re-referenced
+    resample_rate: float or list
+        The resampling frequency
+    dtype: dtype or None
+        The dtype of the returned traces. If None, the dtype of the parent recording is used
+
+    Returns
+    -------
+    resample_recording: ResampleRecording
+        The resampled recording extractor object
+
+    """
+
+    name = "resample"
+    installed = HAVE_RR  # check at class level if installed or not
+    installation_mesg = (
+        "To use the ResampleRecording, install scipy: \n\n pip install scipy\n\n"  # err
+    )
+
+    def __init__(
+        self,
+        recording,
+        resample_rate,
+        dtype=None,
+    ):
+        # Check that scipy is present
+        assert self.installed, self.installation_mesg
+        # Original sampling frequency
+        self._orig_samp_freq = recording.get_sampling_frequency()
+        self._resample_rate = resample_rate
+        self._sampling_frequency = resample_rate
+        dtype = fix_dtype(recording, dtype)
+        self.dtype = dtype
+
+        BasePreprocessor.__init__(
+            self, recording, sampling_frequency=resample_rate, dtype=dtype
+        )
```

Comment:
We have to be careful here because some segments mugh have a `time_vector` (see `BaseRecording.set_times()`). In that case we could either:
- resample the time vector as well
- throw a warning and discard the time vector

Reply:
I just discarded it.... but it works.